### PR TITLE
NOJIRA: Web template access to envFrom

### PR DESCRIPTION
**Pull request by jfc-costa**
In the rcalc amra we would like to have a repo that will configure common configmaps like for example kafka configs; (broker list , security, schema registry url,...).
And with it we would like to use the `envFrom` so we could reference just the configmap instead of mapping one by one.
So the expected would be to have this:
```yaml
  environmentEnvRefs:
    - configMapRef:
        name: rcalc-kafka-config-map
```

instead of:
```yaml
  environmentEnvVars: 
    - name: KAFKA_BROKER_URL
      valueFrom:
        configMapKeyRef:
          name: rcalc-kafka-config-map
          key: KAFKA_BROKER_URL
    - name: SECURITY_PROTOCOL
      valueFrom:
        configMapKeyRef:
          name: rcalc-kafka-config-map
          key: SECURITY_PROTOCOL
    - name: SCHEMA_REGISTRY_URL
      valueFrom:
        configMapKeyRef:
          name: rcalc-kafka-config-map
          key: SCHEMA_REGISTRY_URL
```

Let me know if you have any better name suggestions / concerns

### CODE REVIEW

File: charts/bandstand-web-service/templates/deployment.yaml [hunk 1]
```diff
@@ -114,6 +114,10 @@ spec:
             {{- if .Values.environmentEnvVars }}
               {{- .Values.environmentEnvVars | toYaml | trimPrefix "|" | trimPrefix "\n" | nindent 12 }}
             {{- end }}
+          envFrom:
+            {{- if .Values.environmentEnvRefs }}
+            {{- .Values.environmentEnvRefs | toYaml | trimPrefix "|" | trimPrefix "\n" | nindent 12 }}
+            {{- end }}
```
Comment (gavinclarkeuk) [inline charts/bandstand-web-service/templates/deployment.yaml:120]:
```suggestion
          {{- if .Values.environmentEnvRefs }}
          envFrom:
            {{- .Values.environmentEnvRefs | toYaml | trimPrefix "|" | trimPrefix "\n" | nindent 12 }}
          {{- end }}
```

File: charts/bandstand-web-service/templates/tests/pod.yaml [hunk 2]
```diff
@@ -51,6 +51,11 @@ spec:
         {{- if (.Values.test).environmentEnvVars }}
           {{- .Values.test.environmentEnvVars | toYaml | trimPrefix "|" | trimPrefix "\n" | nindent 8 }}
         {{- end }}
+      envFrom:
+        {{- if .Values.sharedEnvConfig }}
+        {{- .Values.sharedEnvConfig | toYaml | trimPrefix "|" | trimPrefix "\n" | nindent 12 }}
+        {{- end }}
```
Comment (gavinclarkeuk) [inline charts/bandstand-web-service/templates/tests/pod.yaml:57]:
```suggestion
      {{- if .Values.sharedEnvConfig }}
      envFrom:
        {{- .Values.sharedEnvConfig | toYaml | trimPrefix "|" | trimPrefix "\n" | nindent 12 }}
      {{- end }}
```

Comment (jfc-costa) [reply charts/bandstand-web-service/templates/tests/pod.yaml]: thanks, was just looking at it and forgot to update the name and use the test too 🤦

**Comment by gavinclarkeuk** [comment]
Thanks, this looks good. A few additional things:
* can we just call the fields `envFrom` and `test.envFrom`
* could you also apply the change to the other charts to keep them consistent
* will need to bump the chart version in the top level Chart.yaml
* also need to add the fields to the relevant docs for each chart

**Comment by jfc-costa** [comment]
sorry, bank holiday on my side.
I think I applied all the changes but feel free to do any changes that you think are necessary 🙏

**Review by gavinclarkeuk — APPROVED** [review]
Made some minor docs tweaks, but overall this is great. Thanks!

**Pull request merged**